src/sourceMaps/sourceMap.ts

```diff
@@ -132,25 +133,18 @@ export class SourceMap {
         // sourceMap.sources are initially relative paths, file:/// urls, made-up urls like webpack:///./app.js, or paths that start with /.
         // resolve them to file:/// urls, using computedSourceRoot, to be simpler and unambiguous, since
         // it needs to look them up later in exactly the same format.
-        const normalizedSources = sourceMap.sources.map(sourcePath => {
-            if (isDefined(sourceMapPathOverrides)) {
-                const fullSourceEntry = sourceMapUtils.getFullSourceEntry(sourceMap.sourceRoot, sourcePath);
-                const mappedFullSourceEntry = sourceMapUtils.applySourceMapPathOverrides(fullSourceEntry.textRepresentation, sourceMapPathOverrides, isVSClient);
-                if (fullSourceEntry.textRepresentation !== mappedFullSourceEntry) {
-                    return mappedFullSourceEntry; // If we found a path override that applies, return the result of applying it
-                }
-            }
-
-            if (sourcePath.startsWith('file://')) {
-                // strip file://
-                return new LocalFileURL(sourcePath).filePathRepresentation;
-            }
-
-            if (!path.isAbsolute(sourcePath)) {
-                // Overrides not applied, use the computed sourceRoot
-                return path.resolve(computedSourceRoot, sourcePath);
-            } else {
-                return sourcePath;
+        const normalizedSources: string[] = [];
+        sourceMap.sources.forEach(sourcePath => {
+            try {
+                normalizedSources.push(SourceMap.normalizeSource(sourceMapPathOverrides, sourceMap, isVSClient, computedSourceRoot, sourcePath));
+            } catch (exception) {
+                // If we fail to process a single source path, we'll try to process the other ones anyways instead of just failing everything
+                // this is important because we've seen some sourcemap with null source entries
+                telemetry.reportError('Failed to parse sourcemap source', exception);
+
+                // We add the unnormalized source when things fail. Processing this source will most likely fail in other places of the code.
+                // We need to insert something to preserve the indexes of the sources matching the ones for the mappings, etc...
+                normalizedSources.push(`${sourcePath}`);
```

Comment:
Just `normalizedSources.push(sourcePath)`?

Reply:
I want to convert it to a string in case it's null, or anything that is not a string...

Reply:
Oh ok